traffic_monitor/poller/cache.go

```diff
@@ -162,14 +181,30 @@ func poller(
 			pollID := atomic.AddUint64(&pollNum, 1)
 			pollFinishedChan := make(chan uint64)
 			log.Debugf("poll %v %v start\n", pollID, time.Now())
-			bts, reqEnd, reqTime, err := pollFunc(pollCtx, url, host, pollID)
-			rdr := io.Reader(nil)
-			if bts != nil {
-				rdr = bytes.NewReader(bts) // TODO change handler to take bytes? Benchmark?
+			if usingIPv4 {
+				bts, reqEnd, reqTime, err := pollFunc(pollCtx, url, host, pollID)
```

Comment:
It looks like this and the next block are identical, except for `url`. Could the duplication be reduced by changing this to 
```
pollURL := url
if !usingIPv4 {
  pollURL = url6
}
bts, reqEnd, reqTime, err := pollFunc(pollCtx, pollURL, host, pollID)
```
?

Reply:
done